### PR TITLE
Tests/new test cases

### PR DESCRIPTION
**Pull request by vallo87**
I wanted to practice some testing therefore I just randomly covered some missing tests, I hope it was a good job

- AppController
- DashboardController

### CODE REVIEW

**Comment by didoda** [comment]
Well done! I made a review requesting some changes.
I suggest you, after opening a PR, to take some time to re-check your code. Sometimes minor issues can be discovered at that point, in github code comparison view.
Pay attention in phpdoc: `@param` etc.
Check class imports at the beginning of files (`use [...]`) for unused or wrong imports.

File: tests/TestCase/Controller/AppControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+        $updatedToken = [
+            'jwt' => 'oldToken',
+            'renew' => 'this is a new token',
+        ];
+
+        $apiClient = $this->getMockBuilder(BEditaClient::class)
+            ->setConstructorArgs(['https://media.example.com'])
+            ->getMock();
+
+            // mocked getTokens method returns fake tokens
```
Comment (didoda) [inline tests/TestCase/Controller/AppControllerTest.php]: minor: wrong indentation. Remove 4 spaces at the beginning of this line

File: tests/TestCase/Controller/AppControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
+        $event = $this->App->dispatchEvent('Controller.beforeRender');
+        $this->App->beforeRender($event);
+
+        static::assertArrayHasKey('user', $this->App->viewVars);
+        static::assertEquals($user, $this->App->viewVars['user']);
+    }
+
+    /**
+     * Test `beforeRender` method
```
Comment (didoda) [inline tests/TestCase/Controller/AppControllerTest.php]: minor: this method is a different test, compared to the previous; some extra doc could be added.
I.e.:
```
Test `beforeRender` method, when updating tokens in session

File: tests/TestCase/Controller/AppControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+        $flash = $this->App->request->getSession()->read('Flash');
+
+        $expected = __('You are not logged or your session has expired, please provide login credentials');
+        $message = $flash['flash'][0]['message'];
+
+        static::assertEquals($expected, $message);
+    }
+
+    /**
+     * test 'beforeFilter' correct apiClient token setup
```
[the remark this answers is not in the document]
Comment (vallo87) [reply tests/TestCase/Controller/AppControllerTest.php]: i didn't get it

[the remark this answers is not in the document]
Comment (vallo87) [reply tests/TestCase/Controller/AppControllerTest.php]: ah ok backticks..

File: tests/TestCase/Controller/AppControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+     *
+     * @return void
+     */
+    public function testCheckRequest($expected, $params, $config) : void
+    {
+        $this->setupController($config);
+
+        if ($config == null) {
+            $this->App->request = $config;
+        }
```
[the remark this answers is not in the document]
Comment (vallo87) [reply tests/TestCase/Controller/AppControllerTest.php]: it's necessary to test for empty request object if config == null then it sets request to null in order to test the method invoked on the BadRequestException('Empty request')

File: tests/TestCase/Controller/AppControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+use App\Controller\AppController;
+use BEdita\SDK\BEditaClient;
+use BEdita\WebTools\ApiClientProvider;
+use Cake\Core\Configure;
+use Cake\Http\ServerRequest;
+use Cake\Network\Exception\BadRequestException;
+use Cake\Network\Exception\MethodNotAllowedException;
+use Cake\TestSuite\TestCase;
+use Composer\Script\Event;
+use Symfony\Component\Config\Definition\Exception\Exception;
```
[the remark this answers is not in the document]
Comment (vallo87) [reply tests/TestCase/Controller/AppControllerTest.php]: that was the autocomplete... as you said in the message, I should've checked thoroughly ( I actually checked the files of course, but I skipped some lines)

**Comment by vallo87** [comment]
sorry for the tooooo many typos and minors, thankfully you have a good eye

**Pull request merged**